src/main/java/pl/most/typer/controller/footballController/MatchController.java

```diff
@@ -48,39 +52,56 @@ public String updateMatchInfo(@PathVariable(value = "id") Integer id) {
     private String getMatchesByStages(@PathVariable("id") Integer id,
                                       @RequestParam(name = "stage", required = false) String stage,
                                       @RequestParam(name = "TCID", required = false) String typerCompetitionId,
-                                      Model model) {
+                                      Model model) throws ResourceException {
         Optional<Competition> optionalCompetition = Optional.of(competitionService.getCompetition(id));
```

Comment:
competitionService zwraca Competition i moze rzucic wyjatkiem -> to tez by trzeba bylo wrzucic do try catch i nie wkladac tego w Optional.of(). Ja to zrobilem na etapie refactoringu mojego Service zeby u Ciebie dzialalo poprawnie